### PR TITLE
[PLUGIN-1647] fix the access token provider

### PR DESCRIPTION
**Pull request by itsankit-google**
This PR does the following:

- Context: GCS Hadoop connector version was upgraded in #1270.

- Fixes the `ACCESS_TOKEN_PROVIDER_IMPL_SUFFIX`.

- `hadoop-connector` internall stopped propagating the config to `AccessTokenProvider` in `CredentialFromAccessTokenProviderClassFactory#credential` from `2.2.3` and started propagating the config in `GoogleHadoopFileSystemBase#getAccessTokenProvider` which caused `NPE` while creating `BigQueryCredential`. Hence, provided an empty configuration in `ServiceAccountAccessTokenProvider` when `conf` is `null`.

ref: https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/587/files#diff-bef33ce03ad3ec694eec0ef295fc56cb9240e087f4c3773036fe65d820d68037R1323

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProvider.java [hunk 1]
```diff
@@ -56,7 +56,13 @@ public void refresh() throws IOException {
 
   private GoogleCredentials getCredentials() throws IOException {
     if (credentials == null) {
-      credentials = GCPUtils.loadCredentialsFromConf(conf);
+      if (conf != null) {
+        credentials = GCPUtils.loadCredentialsFromConf(conf);
+      } else {
+        // when initializing {@link ForwardingBigQueryFileOutputCommitter}
+        // conf is null.
```
Comment (sahusanket) [inline src/main/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProvider.java]: Please add more details in comments as in why is this necessary .

Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProvider.java]: Done.

**Review by tivv — COMMENTED** [review]
Please add a test

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java [hunk 2]
```diff
@@ -137,6 +142,44 @@ public OutputCommitter createCommitter(TaskAttemptContext context) throws IOExce
     return new BigQueryOutputCommitter(context, delegateCommitter);
   }
 
+  /**
+   * This method is copied from
+   * {@link ForwardingBigQueryFileOutputFormat#checkOutputSpecs(JobContext)} to override
+   * {@link BigQueryFactory} with {@link BigQueryFactoryWithScopes}.
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java:147]: nit: We should raise a ticket against BigQuery connector to fix the scopes

File: src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
```
Comment (tivv) [inline src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java]: nit: 2023

File: src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    Configuration conf = new Configuration();
+    for (Map.Entry<String, String> prop : authProperties.entrySet()) {
+      conf.set(prop.getKey(), prop.getValue());
+    }
+
+    AccessTokenProvider accessTokenProvider =
+        HadoopCredentialConfiguration.getAccessTokenProvider(conf, ImmutableList.of(
+        GoogleHadoopFileSystemConfiguration.GCS_CONFIG_PREFIX));
+
+    Assert.assertTrue(accessTokenProvider instanceof ServiceAccountAccessTokenProvider);
```
Comment (tivv) [inline src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java]: nit: `assertThat(accessTokenProvider, instanceOf(ServiceAccountAccessTokenProvider.class));` It would give proper message when failed instead of just "assertion failed"

**Review by tivv — APPROVED** [review]

File: src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java (hunk 3 above)
Comment (itsankit-google) [reply src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java]: done.

File: src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java (hunk 4 above)
Comment (itsankit-google) [reply src/test/java/io/cdap/plugin/gcp/gcs/ServiceAccountAccessTokenProviderTest.java]: done.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java (hunk 2 above)
Comment (itsankit-google) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java:147]: Actually BigQuery Hadoop Connector support has been discontinued now: https://github.com/GoogleCloudDataproc/hadoop-connectors/pull/930
